ignition/CMakeLists.txt

```diff
@@ -94,7 +101,8 @@ target_link_libraries(GazeboWrapper
     ECMSingleton
     IgnitionRobot
     ignition-gazebo3::core
-    tiny-process-library)
+    tiny-process-library
+    ignition-fuel_tools${IGN_FUEL_TOOLS_VER}::ignition-fuel_tools${IGN_FUEL_TOOLS_VER})
```

Comment:
Ow ok, now I got why the `GymFactory` target does not carry the fuel library. You included some Ignition header in `GazeboWrapper.h`, that is part of the public APIs, but you linked fuel as PRIVATE library. This means that all targets that link agains the `GazeboWrapper` target **do not** get fuel linked. This happens for the `GymFactory` target and is propagated to the examples. 

The solution would be to make fuel a PUBLIC linked library, but it defeats the scope of this wrapper, that is providing a layer to hide all the Ignition components to downstream users.

This change is fine after you address this comment.